viral_seq/tests/test_workflow.py

```diff
@@ -1101,12 +1161,35 @@ def test_pearson_aggregation():
         random_state=random_state,
     )
 
+    feature_rank = feature_count["Features"]
     pearson_rank = feature_count["Pearson R"]
-    pearson_rank_exp = [
-        0.9625443164192421,
-        -0.9455651466704694,
-        0.8684269994231254,
-        0.700139321523472,
-    ]
-    # check first four pearson values, numbers have tendency to vary slightly based on dependency versions
-    assert_allclose(pearson_rank[: len(pearson_rank_exp)], pearson_rank_exp, rtol=6e-6)
+    # only the first four features should be showing
+    # up at all in the top 4, based on synthetic correlation:
+    assert_array_equal(feature_count["Sum"][:4], 8)
+    assert_array_equal(feature_count["Sum"][4:], 0)
+    # expected property of the test is ranking kmer_0 > kmer_2 > kmer_3
+    # based on alignment of features and that
+    # ``kmer_1`` should be ranked lowest because of inverse alignment (-1.0)
+    feature_rank_exp = kmer_names[np.asarray([0, 2, 3, 1])]
+    assert_array_equal(feature_rank[:4], feature_rank_exp)
+    # random sampling used in ``StratifiedKFold`` causes features with
+    # perfect/inverse/moderate alignment to have >0.92
+    # pearson correlation with noisy features having lower correlation
+    assert np.all(np.abs(pearson_rank[:4]) > 0.92)
+    assert_array_less(np.abs(pearson_rank[4:]), 0.89)
+
+
+def test_sort_feature_counts():
+    feature_df = pd.DataFrame()
+    kmer_features = np.array(["kmer_AA_ABCDEFG", "kmer_PC_GCAFGDG", "kmer_PC_0134657"])
+    feature_df["Features"] = kmer_features
+    feature_df["Clfr_0"] = [5, 1, 5]
+    feature_df["SHAP_0"] = [4, 1, 1]
+    feature_df["Clfr_1"] = [1, 6, 3]
+    feature_df["SHAP_1"] = [1, 0, 4]
+    feature_df["Pearson R"] = [0.99, -0.99, -0.41]
+
+    out_df = feature_importance.sort_feature_counts(feature_df, n_folds=5)
+    out_features = np.asarray(out_df["Features"])
+    out_exp = kmer_features[np.asarray([2, 0, 1])]
+    assert_array_equal(out_exp, out_features)
```

Comment:
The order of arguments here is reversed--I mentioned this several times over the past year--the documented function signature is:

`assert_array_equal(actual, desired)`

I'll fix it myself now...